tests/e2e/tests/mixer/mixer_test.go

```diff
@@ -466,7 +466,82 @@ func testDenials(t *testing.T, rule string) {
 	if err := visitProductPage(productPageTimeout, http.StatusOK, &header{"x-user", "testuser"}); err != nil {
 		fatalf(t, "product page was not denied: %v", err)
 	}
+}
+
+func TestIngressCheckCache(t *testing.T) {
+	// Apply denial rule to istio-ingress, so that only request with ["x-user"] could go through.
+	// This is to make the test focus on ingress check cache.
+	t.Logf("block request through ingress if x-user header is missing")
+	if err := applyMixerRule(ingressDenialRule); err != nil {
+		fatalf(t, "could not create required mixer rule: %v", err)
+	}
+	defer func() {
+		if err := deleteMixerRule(ingressDenialRule); err != nil {
+			t.Logf("could not clear rule: %v", err)
+		}
+	}()
+	allowRuleSync()
+
+	promAPI, err := promAPI()
+	if err != nil {
+		fatalf(t, "Could not build prometheus API client: %v", err)
+	}
+
+	prior, err := getCachedCheckCalls(promAPI)
+	if err != nil {
+		fatalf(t, "Unable to retrieve valid cached hit number: %v", err)
+	}
+
+	// Visit product page for 5 times. All of them should be denied.
+	for i := 0; i < 5; i++ {
+		if err = visitProductPage(productPageTimeout, http.StatusForbidden); err != nil {
```

Comment:
Can you refactor common code between `TestIngressCheckCache` and `TestCheckCache` ?
TestEgressCache will also follow at some point, it should be simple to add.

Reply:
Done.